dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/AbstractOracleConnector.java

```diff
@@ -190,15 +176,46 @@ Properties buildProperties(@Nonnull ConnectorArguments arguments) {
   // jdbc:oracle:thin:<TNSName> (from 10.2.0.1.0)
   static String buildUrl(ConnectorArguments arguments) {
     String url = arguments.getUri();
-    if (url == null) {
-      String host = arguments.getHost();
-      int port = arguments.getPort(OPT_PORT_DEFAULT);
-      if (isOracleSid(arguments)) {
-        url = "jdbc:oracle:thin:@" + host + ":" + port + ":" + arguments.getOracleSID();
-      } else {
-        url = "jdbc:oracle:thin:@//" + host + ":" + port + "/" + arguments.getOracleServicename();
-      }
+    String host = arguments.getHost();
+    int port = arguments.getPort(OPT_PORT_DEFAULT);
+    if (url != null) {
+      checkNonUriFlags(arguments);
+      return url;
+    }
+    checkServiceAndSid(arguments);
+    if (arguments.getOracleSID() != null) {
+      return "jdbc:oracle:thin:@" + host + ":" + port + ":" + arguments.getOracleSID();
+    } else {
+      return "jdbc:oracle:thin:@//" + host + ":" + port + "/" + arguments.getOracleServicename();
+    }
+  }
+
+  private static void checkNonUriFlags(ConnectorArguments arguments) {
+    if (arguments.getHost() != null) {
+      throw extraFlagProvided("host");
```

Comment:
I would suggest referencing the flag constant names.

For example,
https://github.com/google/dwh-migration-tools/blob/3fdca7661eb364bbd9c0736852d4d65d40df6124/dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java#L99

It would be even better if `arguments.getHost()` returned both the value and the reference to some metadata structure containing the details about the flag (e.g. `OptionSpec` instance), but that might be a separate big change, so unlikely to be included in this PR.

Reply:
Done, I added references to the ConnectorArgs constants